x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/pages/coverage_overview/filters_panel.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFilterButton, EuiFilterGroup, EuiFlexGroup, EuiFlexItem, EuiPanel } from '@elastic/eui';
+import React, { memo, useCallback } from 'react';
+import { CoverageOverviewLegend } from './shared_components';
+import * as i18n from './translations';
+
+export interface CoverageOverviewFiltersPanelProps {
+  setShowExpandedCells: (arg: boolean) => void;
+  showExpandedCells: boolean;
+}
+
+const CoverageOverviewFiltersPanelComponent = ({
+  setShowExpandedCells,
+  showExpandedCells,
+}: CoverageOverviewFiltersPanelProps) => {
+  const handleExpandCellsFilterClick = useCallback(
+    () => setShowExpandedCells(true),
+    [setShowExpandedCells]
+  );
+  const handleCollapseCellsFilterClick = useCallback(
+    () => setShowExpandedCells(false),
+    [setShowExpandedCells]
+  );
+
+  return (
+    <EuiPanel>
+      <EuiFlexGroup justifyContent="spaceBetween">
+        <EuiFlexItem grow={false}>
+          <EuiFilterGroup>
+            <EuiFilterButton
+              withNext
+              hasActiveFilters={!showExpandedCells}
+              onClick={handleCollapseCellsFilterClick}
+            >
+              {i18n.COLLAPSE_CELLS_FILTER_BUTTON}
+            </EuiFilterButton>
+            <EuiFilterButton
+              hasActiveFilters={showExpandedCells}
+              onClick={handleExpandCellsFilterClick}
+            >
+              {i18n.EXPAND_CELLS_FILTER_BUTTON}
+            </EuiFilterButton>
+          </EuiFilterGroup>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <CoverageOverviewLegend />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </EuiPanel>
+  );
+};
+
+export const CoverageOverviewFiltersPanel = memo(CoverageOverviewFiltersPanelComponent);
```

Comment:
Obviously having all components wrapped in `memo()` and additionally use `useMemo()` in `CoverageOverviewPageComponent` won't improve performance. The main performance hit is reflow and we have to avoid it whenever possible so having an extra render usually might not be an issue.

Ideally we need to compare performance without using `memo` and `useMemo` and with them to see if we improve something as it doesn't work when references aren't stable and change each render.

It should be safe to get rid of `memo()` wrapper around components in the folder.